dipy/reconst/forecast.py

```diff
@@ -0,0 +1,589 @@
+from __future__ import division
+
+from warnings import warn
+import numpy as np
+from dipy.reconst.cache import Cache
+from dipy.reconst.multi_voxel import multi_voxel_fit
+from dipy.reconst.shm import real_sph_harm
+from dipy.core.gradients import gradient_table
+from scipy.special import erf
+from math import factorial
+from dipy.core.geometry import cart2sphere
+from dipy.data import get_sphere
+from dipy.reconst.odf import OdfModel, OdfFit
+from scipy.optimize import leastsq
+from dipy.utils.optpkg import optional_package
+cvxpy, have_cvxpy, _ = optional_package("cvxpy")
+
+
+class ForecastModel(OdfModel, Cache):
+    r"""Fiber ORientation Estimated using Continuous Axially Symmetric Tensors 
+    (FORECAST) [1,2,3]_. FORECAST is a Spherical Deconvolution reconstruction model
+    for multi-shell diffusion data which enables the calculation of a voxel
+    adaptive response function using the Spherical Mean Tecnique (SMT) [2,3]_. 
+
+    With FORECAST it is possible to calculate crossing invariant parallel
+    diffusivity, perpendicular diffusivity, mean diffusivity, and fractional
+    anisotropy [2]_
+
+    References
+    ----------
+    .. [1] Anderson A. W., "Measurement of Fiber Orientation Distributions
+           Using High Angular Resolution Diffusion Imaging", Magnetic
+           Resonance in Medicine, 2005.
+
+    .. [2] Kaden E. et. al, "Quantitative Mapping of the Per-Axon Diffusion 
+           Coefficients in Brain White Matter", Magnetic Resonance in 
+           Medicine, 2016.
+
+    .. [3] Zucchelli E. et. al, "A generalized SMT-based framework for
+           Diffusion MRI microstructural model estimation", MICCAI Workshop
+           on Computational DIFFUSION MRI (CDMRI), 2017.
+    Notes
+    -----
+    The implementation of FORECAST may require CVXPY (http://www.cvxpy.org/).
+    """
+
+    def __init__(self,
```

Comment:
Ok, I see, there is the same mistake on SHORE and MAPMRI :-).  `ReconstModel ` should initialize some variable (like gtab). You can see some examples on FWDTI, LIFE or SFM.

You should keep your  `__init__`, this one initialize your class `ForecastModel`. You just have to initialize `ReconstModel ` since you inherit from this class. 
```
def __init__(self, gtab, sh_order=8, lambda_lb=1e-3,
             optimizer='wls', sphere=None, lambda_csd=1.0)

        # your docstrings
        super(ForecastModel, self).__init__(gtab)  # or ReconstModel.__init__(gtab) 
```